arm/examples/sdk.go

```diff
@@ -0,0 +1,168 @@
+	SubscriptionID string
+
+	// TenantID is the `tenant` from an Azure Service Principal.
+	TenantID string
+
+	// HashMap is the data value some parts of the SDK will expect, and is stored here as convenience.
+	HashMap map[string]string
+
+	// AuthenticatedToken is the generated token that is needed to authenticate with each client.
+	AuthenticatedToken *adal.ServicePrincipalToken
+}
+
+func NewSDK() (*SDK, error) {
+
+	// -------------------------------------------------------------------
+	// Validation that the environment is configured correctly.
+	//
+	clientID := os.Getenv("AZURE_CLIENT_ID")
+	if clientID == "" {
+		return nil, fmt.Errorf("empty $AZURE_CLIENT_ID")
+	}
+	clientSecret := os.Getenv("AZURE_CLIENT_SECRET")
+	if clientSecret == "" {
+		return nil, fmt.Errorf("empty $AZURE_CLIENT_SECRET")
+	}
+	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	if subscriptionID == "" {
+		return nil, fmt.Errorf("empty $AZURE_SUBSCRIPTION_ID")
+	}
+	tenantID := os.Getenv("AZURE_TENANT_ID")
+	if tenantID == "" {
+		return nil, fmt.Errorf("empty $AZURE_TENANT_ID")
+	}
+
+	// Here we build the SDK based on a user's configuration.
+	sdk := &SDK{
+		Authentication: &Authentication{
+			ClientID:       clientID,
+			ClientSecret:   clientSecret,
+			SubscriptionID: subscriptionID,
+			TenantID:       tenantID,
+			HashMap: map[string]string{
+				"AZURE_CLIENT_ID":       clientID,
+				"AZURE_CLIENT_SECRET":   clientSecret,
+				"AZURE_SUBSCRIPTION_ID": subscriptionID,
+				"AZURE_TENANT_ID":       tenantID,
+			},
+		},
+	}
+
+	// Generate an authenticated token.
+	authenticatedToken, err := helpers.NewServicePrincipalTokenFromCredentials(sdk.Authentication.HashMap, azure.PublicCloud.ResourceManagerEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	sdk.Authentication.AuthenticatedToken = authenticatedToken
+	return sdk, nil
+}
+
+// E will check if an error is nil, and exit the program in the case of an error.
+// Use this as convenience to simplify code in various examples.
+func E(err error) {
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+		os.Exit(2)
+	}
+}
+
+// S will take an arbitrary string, and return a newly created pointer to this string.
+// Use this as convenience to simplify code in various examples.
+func S(i string) *string {
+	return &i
+}
+
+// I32 will take an arbitrary int, and return a newly created pointer to an int32.
+// Use this as convenience to simplify code in various examples.
+func I32(i int) *int32 {
+	i32 := int32(i)
+	return &i32
+}
+
+// F64 will take an arbitrary int, and return a newly created pointer to a float64.
+// Use this as convenience to simplify code in various examples.
+func F64(i int) *float64 {
+	f64 := float64(i)
+	return &f64
+}
+
+// PrettyJSON will take an arbitrarry interface{} and attempt to print the data structure out to STDOUT.
+// Use this as convenience to simplify code in various examples.
+func PrettyJSON(data interface{}) error {
```

Comment:
In general, I'm not a big fan of functions that print straight to STDOUT. Mind if we return the Prettified JSON instead? It should be easy enough for the caller to print as they see appropriate.